internal/provider/resources/deployment.go

```diff
@@ -762,7 +764,7 @@ func (r *DeploymentResource) Update(ctx context.Context, req resource.UpdateRequ
 		WorkQueueName:          model.WorkQueueName.ValueString(),
 	}
 
-	if !model.ConcurrencyOptions.CollisionStrategy.IsNull() {
+	if model.ConcurrencyOptions != nil {
```

Comment:
this wasn't causing problems, but in `Create()` I noticed that we do a nil check on the parent `ConcurrencyOptions` pointer struct, rather than the nested `CollisionStrategy` attribute (since that won't scale if we need to add more attributes).  So i updated this here to match

Reply:
good catch, I missed that